var/spack/repos/builtin/packages/autoconf/package.py

```diff
@@ -18,17 +18,30 @@ class Autoconf(AutotoolsPackage, GNUMirrorPackage):
             preferred=True)
     version('2.62', sha256='83aa747e6443def0ebd1882509c53f5a2133f502ddefa21b3de141c433914bdd')
     version('2.59', sha256='9cd05c73c5fcb1f5ccae53dd6cac36bb8cb9c7b3e97ffae5a7c05c72594c88d8')
-    version('2.13', sha256='f0611136bee505811e9ca11ca7ac188ef5323a8e2ef19cffd3edb3cf08fd791e')
 
     # https://savannah.gnu.org/support/?110396
     patch('https://git.savannah.gnu.org/cgit/autoconf.git/patch/?id=05972f49ee632cd98057a3caf82ebfb9574846da',
           sha256='eaa3f69d927a853313a0b06e2117c51adab6377a2278549b05abc5df93643e16',
           when='@2.70')
+    # tests/foreign.at (Libtool): Be tolerant of 'quote' replacing the older `quote'
+    patch('http://mirrors.mit.edu/gentoo-portage/sys-devel/autoconf/files/autoconf-2.69-fix-libtool-test.patch',
+          sha256='7793209b33013dc0f81208718c68440c5aae80e7a1c4b8d336e382525af791a7',
+          when='@2.69')
+    # Fix autoscan when using Perl 5.17+ (reported already in January 2013)
+    patch('http://mirrors.mit.edu/gentoo-portage/sys-devel/autoconf/files/autoconf-2.69-perl-5.26.patch',
+          sha256='35c449281546376449766f92d49fc121ca50e330e60fefcfc9be2af3253082c2',
+          when='@2.62:2.69')
```

Comment:
If this only applies for `perl@5.17:` then you can say `when=@2.62:2.69 ^perl@5.17:`.

Reply:
Cool that it is possible to do that.

But, in reality, the only perl older than 5.17 which spack has is 5.16.3, and it's build is broken (it's from 2013):

```py
     4187    ../../miniperl -I../../lib Makefile.PL INSTALLDIRS=perl INSTALLMAN1DIR=none INSTALLMAN3DIR=none PERL_CORE=1 LIBPERL_A=libperl.s
             o LINKTYPE=dynamic
     4188    INCLUDE & LIB must be the same when BUILD_ZLIB is True
     4189    Parsing config.in...
     4190    Writing Makefile for Devel::Peek
     4191    65280 from cpan/Compress-Raw-Zlib's Makefile.PL at make_ext.pl line 409.
     4192    Warning: No Makefile!
  >> 4193    make[1]: *** No rule to make target 'config'.  Stop.
```
`grep -r perl@5.16.3` does not found references to perl@5.16.3, so we should remove it, ok?

For now, I applied your suggestion, since it it is a theoretical consideration.

Could I add the removal of perl version 5.16 (again, it does not build and is from 2013) to this pull request?